Simbody/Visualizer/simbody-visualizer/simbody-visualizer.cpp

```diff
@@ -2043,8 +2043,12 @@ static void readDataFromPipe(int srcPipe, unsigned char* buffer, int bytes) {
     int totalRead = 0;
     while (totalRead < bytes) {
         auto retval = READ(srcPipe, buffer + totalRead, bytes - totalRead);
+        int attempts = 0;
+        while (retval == -1 && errno == 4 && ++attempts < 5) {
+          retval = READ(srcPipe, buffer + totalRead, bytes - totalRead);
```

Comment:
Nitpick: we might be able to avoid duplicating this line using a `do-while` loop:

```cpp
int attempts = 0;
int retval = -1;
do {
    retval = READ(srcPipe, buffer + totalRead, bytes - totalRead);
} while (retval == -1 && errno == 4 && ++attempts < 5);
```

Reply:
agree, pushed this change